File: Java/TelemetrySystem/src/test/java/tddmicroexercises/telemetrysystem/TelemetryClientTest.java
package tddmicroexercises.telemetrysystem;

import org.junit.Assert;
import org.junit.jupiter.api.Test;

import static tddmicroexercises.telemetrysystem.TelemetryClient.DIAGNOSTIC_MESSAGE;

public class TelemetryClientTest {
    ConnectionClass connection = new ConnectionClass();
    ClientClass client = new ClientClass();

    boolean onlineStatus;
    @Test
    void testConnection(){
        connection.connect("connectionstring");
        Assert.assertEquals(false,onlineStatus);
    }
    @Test
    void testDisconnection(){
        connection.disconnect();
        Assert.assertEquals(false, onlineStatus);
    }
    @Test
    void testSend(){
        client.send("AT#UD");
        Assert.assertEquals("AT#UD",DIAGNOSTIC_MESSAGE);
    }
    @Test
    void testReceive(){
       String msg=  client.receive();
       Assert.assertNotEquals(msg,client.receive());

    }
    @Test
    void testGetOnlineStatus(){
       onlineStatus= client.getOnlineStatus();
       Assert.assertEquals(onlineStatus,client.getOnlineStatus());
    }

}

File: kotlin/TelemetrySystem/src/test/kotlin/tddmicroexercises/telemetrysystem/TelemetryClientTest.kt
package tddmicroexercises.telemetrysystem

import org.junit.Assert
import org.junit.Test
import java.util.*

class TelemetryClientTest {

    inner class StubRandom : Random() {
        override fun nextInt(bound: Int): Int {
            return 9
        }
    }

    @Test
    fun `Should connect successfully`() {
        val telemetryClient = TelemetryClient()
        telemetryClient.connect("testing")
        Assert.assertTrue(telemetryClient.isOnline)
    }

    @Test
    fun `Should not connect successfully`() {
        val telemetryClient = TelemetryClient(StubRandom())
        telemetryClient.connect("testing")
        Assert.assertFalse(telemetryClient.isOnline)
    }
}